### ISSUE TITLE
korc file

### ISSUE THREAD

**Issue by mattmoor** [google/ko#143]
It would be useful to be able to have a `.korc` file akin to a `.bazelrc` file that allows for certain flags to be passed by default.

For example, passing `-B` or `-P` gets tedious, and if I exclusively use a registry that supports deep nesting like GCR, I may want `-P` to avoid the noisy hash-based names we use by default.  For a registry like ECR, where I have to create repositories manually before pushing, passing `-B` all the time may be necessary for the push to succeed.  Another good example of this might be if I want to always pass `--strict`.

Initially I'd though `KO_FLAGS=-B`, as we do in our e2e tests, but some of these flags are context-specific (e.g. `-B` won't work with `ko delete`).

Bazel's solution to this is to allow a file in your home directory (and a few other contexts) where you can specify per-subcommand flags, e.g.

```
apply -B --strict
# For releases, use full paths
resolve -P --strict
delete --ignore-not-found
```

Wanted to put this out into the world in case someone was sufficiently motivated to implement it 😇

**Comment by jonjohnsonjr** [comment]
Yeah I'd love this and I'd love to deprecate the equivalent flags. Naming convention is per user not per project.
